streams/src/test/java/org/apache/kafka/streams/processor/internals/ProcessorTopologyTest.java

```diff
@@ -484,6 +517,7 @@ private void assertNextOutputRecord(final String topic,
                                         final Headers headers,
                                         final Integer partition,
                                         final Long timestamp) {
+        @SuppressWarnings("deprecation") //Current public interface support only reading TestRecord, not containing partition
```

Comment:
This is still using deprecated pipeInput because there is partition verification.
Can we skip partition assertion here or do we need some other way to get ProducerRecord?

Reply:
It seems like the purpose of this test class is to verify the test driver. Since the partition is effectively not part of the test driver's (non-deprecated) API, I think we can drop it.

Technically, as long as the deprecated interface is still in the public API, we should still exercise it, but I think in this case it's extremely unlikely we would break the partition field in the future, and it's probably not that risky anyway.

I'd vote just to skip the partition assertion here.

Reply:
Moved to use TestOutputTopic and partition assertion removed.